src/software/ai/navigator/navigator_private.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/obstacle/obstacle_generation.h"
+#include "software/geom/util.h"
+#include "software/new_geom/point.h"
+#include "software/util/parameter/dynamic_parameters.h"
+#include "software/world/world.h"
+
+/**
+ * This files contains private navigator functions that navigator uses
+ * but requires testing
+ */
+
+/**
+ * Calculates the transition speed for the robot between two line segments
+ *
+ * Calculates the speed that the robot should be at when it is at the end of a
+ * given line segment in order to smoothly transition to another given line segment,
+ * given a final speed at the end of the two line segments
+ *
+ * @param p1, p2, p3 are 3 points that define two line segments that form a path
+ * @param final_speed is the intended final speed at the end of the path
+ * @return the first segment's final speed after travelling from p1 to p2
+ * for a smooth transition to the p2 to p3 path, scaled by the final speed at the end
+ * of the path
+ */
+double calculateTransitionSpeedBetweenSegments(const Point &p1, const Point &p2,
```

Comment:
do we need this to be in a file called `navigator_private`? Can we just make a private function of the navigator class?

Reply:
We want to test this function so one way to do that is to create this navigator_private file. Let me know if you have better (and hopefully simple) ideas

Reply:
What about making all of the functions protected, and then creating a subclass in the test file and then test through that? You would still have encapsulation and we wouldn't have to create this separate file just for testing sake? 

~~~c++ from SO
class ToBeTested
{
protected:
    bool SensitiveInternal(int p1, int p2); // Still needs testing
}

// Google-test:
class ToBeTestedFixture : public ToBeTested, public testing::Test
{
   // Empty - bridge to protected members for unit-testing
}

TEST_F(ToBeTestedFixture, TestSensitive)
{
    ASSERT_TRUE(SensitiveInternal(1, 1));
    ASSERT_FALSE(SensitiveInternal(-1, -1));
}
~~~
 
It seems like `navigator_private` is going to become another `util` file and we going to collect a bunch of functions here, IMO we should try to avoid that if possible

Reply:
:arrow_double_up: I much prefer this solution for exactly the reasons that Akhil noted here. No other comments otherwise.

Reply:
+1 for this

Reply:
changed!